src/swap/components/SwapSettings.tsx

```diff
@@ -0,0 +1,130 @@
+import { useCallback, useState } from 'react';
+import {
+  background,
+  cn,
+  color,
+  pressable,
+  text as themeText,
+} from '../../styles/theme';
+import { useBreakpoints } from '../../useBreakpoints';
+import { useIcon } from '../../wallet/hooks/useIcon';
+import type { SwapSettingsReact } from '../types';
+
+export function SwapSettings({
+  className,
+  icon = 'swapSettings',
+  text = '',
+}: SwapSettingsReact) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [slippageMode, setSlippageMode] = useState<'Auto' | 'Custom'>('Auto');
+  const [customSlippage, setCustomSlippage] = useState('0.5');
+  const breakpoint = useBreakpoints();
+
+  const handleToggle = useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen]);
+
+  const iconSvg = useIcon({ icon });
+
+  if (!breakpoint) {
+    return null;
+  }
+
+  // Placeholder for SwapSettingsBottomSheet
+  // Implement mobile version here, similar to WalletBottomSheet
+  if (breakpoint === 'sm') {
+    return <div>Mobile version not implemented</div>;
+  }
+
+  return (
+    <div
+      className={cn(
+        background.default,
+        className,
+        'flex w-full items-center justify-end space-x-1',
+      )}
+      data-testid="ockSwapSettings_Settings"
+    >
+      <span className={themeText.body}>{text}</span>
+      <div className="relative">
```

Comment:
Interesting...

I think we need to reflect on how to structure the whole settings experience.

Like how the Swap components experience will play out
```
<SwapSettings>
  other components
</SwapSettings>
```

@cpcramer this is a good system design challenge for you. This is usally what @abcrane123 really shines, and something I want you to master as well.

How OnchainKit component design pattern makes everything feel simple.

Curios to see how this will play out.

Reply:
Actively refactoring and simplifying styling / removing unnecessary elements

Reply:
Learning a lot as I continue to refactor this...I think the design should work well. It's similar to our other dropdowns + how we are using the WalletBottomSheet. 

I'll should have something ready for review tomorrow morning (missing tests and the SwapSettingsBottomSheet)

Reply:
`isOpen` will likely need to be moved "up" a layer to Swap.tsx to enable clicking off the dropdown to close.

Reply:
sounds good, take your time